### PR TITLE
feat: ignore certain libraries by regex

### LINKED ISSUE #42: Option to exclude imports from beeing resolved
There doesn't seem to be an option to exclude any imports from beeing resolved.
We are building a library which depends on angular material and we provide our own theme.

Therefore we have an import like this in our styles.scss which is later re-used:
```scss
@import '~@angular/material/theming';
```

Now what we want to accomplish is, that all "our" styles are actually bundled into one big scss file, but external style files are just beeing kept as imports so that they can be imported later on in the actual application (which also has a reference to angular material and therefor can resolve it).

### PR DESCRIPTION
**Pull request by cgatian**
Import content can be configured to be ignored with a given regex

This allows library creators to exclude dependent frameworks in their resulting scss file

Closes #42

### CODE REVIEW

**Comment by cgatian** [comment]
@DovydasNavickas / maintainers: I took a stab at this one. Any advice or assistance would be appreciated.

Outstanding problems:
* Unit test- In order to write a unit test, the test bed is going to need modification to allow passing arguments
* General verification of the approach

**Comment by pfeigl** [comment]
Some remarks: 
* you probably should also add an option to arguments.ts for the option (actually without trying it sofar, I'm not sure wether your new code is actually reachable without the option)
* Naming is currently inconsistent: ignoredPaths vs ignoredImports vs ignoredImportRegEx (missing the "s")
* I like the change with the tildeImporter but I guess it should be part of it's own PR, or does the change about the ignoredPaths actually need this?

**Comment by cgatian** [comment]
Great suggestions. 

1)will take another look
2) I was unsure the name of this argument. I didn't want to call it Exclude because it doesn't actually exclude the imports, it just ignores resolution. I'll tighten this up
3)tilde importer is needed. node-sass doesn't support imports with tilde (it's a webpack thing) so this is needed in order to properly resolve the path to node_modules to verify the bundled scss works.

**Comment by cgatian** [comment]
@pfeigl Everything should be resolved now. You can easily test by setting the version of scss-bundle to my commit hash in `package.json`.  (Thanks @zbream for the suggestion it works! 😃 )
```json
"scss-bundle": "git://github.com/cgatian/scss-bundle.git#8bdc623b"
```

**Comment by pfeigl** [comment]
Got a chance to try it out and works as expected. I've got one more remark: When having Verbose logging active, you get something like the following as output:

```
Imports tree:
index.scss
├─┬ theme.scss
│ ├── ..\..\node_modules\@angular\material\_theming.scss
│ └── palette.scss
├── logo.scss
└── ..\external\devextreme\devextreme.scss
```

However in the final result the angular file was actually not included. Therefor you should probably either remove it from the output or add something like `[IGNORED]` to the line. 
The code handling this part already adds things like `[NOT FOUND]` or `[DEDUPED]`.

Thanks for your wonderful changes though 👍

File: src/bundler.ts [hunk 1]
```diff
@@ -106,7 +124,8 @@ export class Bundler {
                 tilde: tilde,
                 path: importName,
                 fullPath: fullPath,
-                found: false
+                found: false,
+                ignored
```
Comment (MartynasZilinskas) [inline src/bundler.ts]: Please use like this:
```
ignored: ignored
```

File: src/bundler.ts [hunk 2]
```diff
@@ -70,7 +79,13 @@ export class Bundler {
...
     }
 
-    private async bundle(filePath: string, content: string, dedupeFiles: string[], includePaths: string[]): Promise<BundleResult> {
+    private async bundle(
+        filePath: string,
+        content: string,
+        dedupeFiles: string[],
+        includePaths: string[],
+        ignoredImports: RegExp[]
+    ): Promise<BundleResult> {
```
Comment (MartynasZilinskas) [inline src/bundler.ts]: Maybe we should update arguments to an `options` variable? 
That will be a breaking change.

/cc @DovydasNavickas

File: src/bundler.ts [hunk 3]
```diff
@@ -91,6 +106,9 @@ export class Bundler {
                 importName += FILE_EXTENSION;
             }
 
+            // Determine if import should be ignored
+            const ignored = ignoredImports.findIndex(ignoredImportRegex => ignoredImportRegex.test(importName)) > -1;
```
Comment (MartynasZilinskas) [inline src/bundler.ts]: Change from this:
```
> - 1
```
to this:
```
!== -1
```

File: src/bundler.ts [hunk 4]
```diff
@@ -175,32 +194,41 @@ export class Bundler {
                     filePath: imp.fullPath,
                     tilde: imp.tilde,
                     found: true,
-                    imports: childImports
+                    imports: childImports,
```
Comment (MartynasZilinskas) [inline src/bundler.ts]: Remove unnecessary comma.

File: tests/cases/ignored-imports/additional.scss [hunk 5]
```diff
@@ -0,0 +1,7 @@
+@import '~@angular/material/theming';
+
+@include mat-core();
+
+.test {
+  color: red;
+}
```
Comment (MartynasZilinskas) [inline tests/cases/ignored-imports/additional.scss]: Add missing new line at the end of the file.

**Comment by MartynasZilinskas** [comment]
It looks like `jest` hangs after successful test run. @cgatian Does jest tests pass locally?

File: src/bundler.ts (hunk 2 above)
Comment (cgatian) [reply src/bundler.ts]: This is probably a really good idea. Personally it's not something I'd want to cover in this PR.

**Comment by cgatian** [comment]
I tried to look into that, running `npm run test-ci` locally works on windows. The TravisCI is linux so I thought maybe that had something to do with it?

File: src/bundler.ts (hunk 4 above)
Comment (cgatian) [reply src/bundler.ts]: Sorry its a habit for me in my stuff to include trailing comma. Removing

**Comment by MartynasZilinskas** [comment]
@cgatian That's strange. I opened a new PR and test succeed. What OS are you using?

**Comment by cgatian** [comment]
@MartynasZilinskas updated. I had to include a change to `gitignore` to allow the `node_modules` directory  within the tests path. Im guessing the tests with `node_modules` were a part of an initial checkin or something?  Not really sure. Anyways, thats the big change I made to this PR outside of your suggestions.

**Comment by cgatian** [comment]
Side request. Could you cut another beta release when this is merged so we can try it for realzies?

**Comment by MartynasZilinskas** [comment]
@cgatian Yeah sure. I will release under `next` tag.

**Pull request merged**